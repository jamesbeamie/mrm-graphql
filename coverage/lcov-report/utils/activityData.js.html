<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for utils/activityData.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">utils</a> activityData.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>1/1</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">&nbsp;
const meetings = {
  DummyDailyRoomEvents: [
    {
      day: 'Apr 29 2019',
      events: [
        {
          cancelled: false,
          eventSummary: 'Kimame Davis sun down Sync',
          startTime: '07:45:00',
          roomName: 'Amara',
          endTime: '08:45:00',
          noOfParticipants: 6,
        },
        {
          cancelled: false,
          eventSummary: 'James Wafula PR review Sync',
          startTime: '07:45:00',
          roomName: 'Amara',
          endTime: '08:45:00',
          noOfParticipants: 6,
        },
        {
          cancelled: true,
          eventSummary: 'James Wafula PR review Sync',
          startTime: '07:45:00',
          roomName: 'Amara',
          endTime: '08:45:00',
          noOfParticipants: 6,
        },
      ],
    },
    {
      day: 'Mon Mar 11 2019',
      events: [
        {
          cancelled: false,
          eventSummary: 'Olayemi Lawal birthday Sync',
          startTime: '07:45:00',
          roomName: 'Amara',
          endTime: '08:45:00',
          noOfParticipants: 6,
        },
      ],
    },
&nbsp;
    {
      day: 'Apr 28 2019',
      events: [
        {
          cancelled: false,
          eventSummary: 'MrUganda Sync',
          startTime: '07:45:00',
          roomName: 'Amara',
          endTime: '08:45:00',
          noOfParticipants: 6,
        },
      ],
    },
    {
      day: 'Apr 28 2019',
      events: [
        {
          cancelled: false,
          eventSummary: 'Bryce Lms Sync',
          startTime: '07:45:00',
          roomName: 'Amara',
          endTime: '08:45:00',
          noOfParticipants: 6,
        },
      ],
    },
  ],
};
export default meetings;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu May 02 2019 21:44:28 GMT+0300 (East Africa Time)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
