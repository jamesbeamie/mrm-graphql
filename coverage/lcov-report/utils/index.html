<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for utils</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> utils
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">93.27% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>97/104</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80.95% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>17/21</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">88% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>22/25</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">94.9% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>93/98</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="Constants.js"><a href="Constants.js.html">Constants.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

<tr>
	<td class="file high" data-value="Cookie.js"><a href="Cookie.js.html">Cookie.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="25" class="abs high">25/25</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="23" class="abs high">23/23</td>
	</tr>

<tr>
	<td class="file high" data-value="Utilities.js"><a href="Utilities.js.html">Utilities.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="40" class="abs high">40/40</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="40" class="abs high">40/40</td>
	</tr>

<tr>
	<td class="file high" data-value="activityData.js"><a href="activityData.js.html">activityData.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

<tr>
	<td class="file high" data-value="checkDate.js"><a href="checkDate.js.html">checkDate.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

<tr>
	<td class="file medium" data-value="formatSetupData.js"><a href="formatSetupData.js.html">formatSetupData.js</a></td>
	<td data-value="72" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 72%;"></div><div class="cover-empty" style="width:28%;"></div></div></td>
	<td data-value="72" class="pct medium">72%</td>
	<td data-value="25" class="abs medium">18/25</td>
	<td data-value="55.56" class="pct medium">55.56%</td>
	<td data-value="9" class="abs medium">5/9</td>
	<td data-value="62.5" class="pct medium">62.5%</td>
	<td data-value="8" class="abs medium">5/8</td>
	<td data-value="78.26" class="pct medium">78.26%</td>
	<td data-value="23" class="abs medium">18/23</td>
	</tr>

<tr>
	<td class="file high" data-value="history.js"><a href="history.js.html">history.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	</tr>

<tr>
	<td class="file high" data-value="notification.js"><a href="notification.js.html">notification.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

<tr>
	<td class="file high" data-value="roomSetupMock.js"><a href="roomSetupMock.js.html">roomSetupMock.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

<tr>
	<td class="file high" data-value="routes.js"><a href="routes.js.html">routes.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu May 02 2019 21:44:28 GMT+0300 (East Africa Time)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
