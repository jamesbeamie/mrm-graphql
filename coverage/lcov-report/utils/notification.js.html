<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for utils/notification.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">utils</a> notification.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/3</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">63x</span>
<span class="cline-any cline-yes">90x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">90x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">/* eslint-disable */
/**
 * toastr =&gt; This is the toastr module imported from 'toastr' npm package
 * notifyType =&gt; This is the notification type (success or error)
 * message =&gt; This is the notification message
 *
 * How to use this function
 * const notify = notification(toastr, 'success', 'noficationMessage');
 * notify()
 */
const notification = (toastr, notifyType, message) =&gt; {
  toastr['options'] = {
    'closeButton': true,
    'positionClass': 'toast-top-center',
    'preventDuplicates': true,
    'onclick': null,
    'showDuration': '300',
    'hideDuration': '1000',
    'timeOut': '2000',
    'extendedTimeOut': '100',
    'showEasing': 'swing',
    'hideEasing': 'linear',
    'showMethod': 'fadeIn',
    'hideMethod': 'fadeOut',
    "newestOnTop": true,
  };
  return () =&gt; toastr[notifyType](message);
};
&nbsp;
export default notification; 
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu May 02 2019 21:44:28 GMT+0300 (East Africa Time)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
