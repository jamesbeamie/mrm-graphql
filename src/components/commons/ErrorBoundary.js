import React from 'react';
import PropTypes from 'prop-types';
import '../../assets/styles/errorBoundary.scss';


/**
 * Error catching component
 *
 * @extends React.Component
 *
 * @returns {JSX}
 */


class ErrorBoundary extends React.Component {
  state = { errorInfo: '', error: '' };

  componentDidCatch(error, errorInfo) {
    this.setState({
      errorInfo,
      error,
    });
  }


  render() {
    if (this.state.errorInfo || this.state.error) {
      return (
        <div>
          {(!this.props.isAuthError || this.state.error) && (this.props.error || this.state.error) }
        </div>
      );
    }
    return this.props.children;
  }
}

ErrorBoundary.propTypes = {
  children: PropTypes.element.isRequired,
  isAuthError: PropTypes.bool,
  error: PropTypes.string,
};

ErrorBoundary.defaultProps = {
  isAuthError: false,
  error: 'Something went wrong',
};

export default ErrorBoundary;
