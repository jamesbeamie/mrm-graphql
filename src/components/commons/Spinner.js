import React from 'react';
import '../../assets/styles/spinner.scss';

/**
 * Reusable Spinner Component
 *
 * @returns {JSX}
 */
const Spinner = prop => (
  <div className="centered">
    <div className={prop.size ? `spinner ${prop.size}` : 'spinner'} />
  </div>
);

Spinner.defaultProps = {
  size: '',
};

export default Spinner;
