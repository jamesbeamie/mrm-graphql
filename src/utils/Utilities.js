/**
 * This file has utility methods that implement ungrouped functionality in the application
 */

/**
 * Converts a url query string into an object containing key value pairs of the query params
 * @param {string} urlQueryString The URL query string e.g ?q=hello&v=waw. Can parse
 * the string got from props.location.search when using react router.
 */
const parseQueryString = (urlQueryString) => {
  // replace first ? with a space
  const formattedQueryString = decodeURIComponent(urlQueryString.replace(/^\?/, ''));
  return formattedQueryString
    .split('&')
    .reduce((parsedObject, stringToSplit) => {
      const [key, value] = stringToSplit.split('=');
      return { ...parsedObject, [key]: value };
    }, {});
};

/**
 * Gets the value of a particular key from localStorage
 * @param {string} itemKey The key of the item to get from local storage
 */
const getItemFromLocalStorage = (itemKey) => {
  try {
    return JSON.parse(localStorage.getItem(itemKey));
  } catch (e) {
    return null;
  }
};

/**
 * Remove the value of a particular key from localStorage
 * @param {string} itemKey The key of the item to remove from local storage
 */
const removeItemFromLocalStorage = (itemKey) => {
  try {
    localStorage.removeItem(itemKey);
    return true;
  } catch (e) {
    return null;
  }
};

/**
 * Stores a an item in the local storage
 * @param {string} key The Key of the item in local storage
 * @param {any} value The value of the item to be store in local storage
 */
const saveItemInLocalStorage = (key, value) => {
  try {
    localStorage.setItem(key, JSON.stringify(value));
    return true;
  } catch (e) {
    return null;
  }
};

/*
* Returns the start and end dates of a 5 day working week
* endDate is either a Friday or current day if the week
* days are still less or equal to Friday
* @retuns {object} that contains startDay and endDay
*/
function thisWeek() {
  const today = Date();
  const date = new Date();
  let milliSecs;
  let weekStart;
  let weekEnd;
  const days = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'];
  const weekDay = today.toString().substring(0, 3);
  if (days.indexOf(weekDay) > 4) {
    milliSecs = date.setDate(date.getDate() - (days.indexOf(weekDay) - 4));
    const fourDays = 4 * 24 * 60 * 60 * 999;
    weekStart = new Date(date - fourDays).toString().substring(4, 15);
    weekEnd = new Date(milliSecs).toString().substring(4, 15);

    return { weekStart, weekEnd };
  }
  milliSecs = date.setDate(date.getDate() - days.indexOf(weekDay));
  weekStart = new Date(milliSecs).toString().substring(4, 15);
  weekEnd = today.toString().substring(4, 15);
  return { weekStart, weekEnd };
}
/**
 * getFirstDayOfTheMonth : This returns the firstDay of the current month.
 * getTodaysDate : This returns today's current date.
 * The format is day/month/year, e.g "01 Nov 2018"
 * @type {Date}
 */
const date = new Date();
const getFirstDayOfTheMonth = () => {
  const firstOfMonth = `${date
    .toString()
    .substring(4, 7)} 01 ${date.toString().substring(11, 15)}`;
  return firstOfMonth;
};
const getTodaysDate = () => {
  const today = date.toString().substring(4, 15);
  return today;
};

export const compressArray = (original) => {
  let compressed = [];
  // make a copy of the input array
  const copy = [...original];

  // first loop goes over every element
  original.forEach((originalItem) => {
    let myCount = 0;
    // loop over every element in the copy and see if it's the same
    copy.forEach((copyItem, i) => {
      if (originalItem === copyItem) {
        // increase amount of times duplicate is found
        myCount += 1;
        // sets item to undefined
        delete copy[i];
      }

      if (myCount > 0) {
        const a = {};
        a.value = original[i];
        a.count = myCount;
        compressed = compressed.filter(({ value }) => value !== copyItem);
        compressed.push(a);
      }
    });
  });
  return compressed;
};


export {
  parseQueryString,
  getItemFromLocalStorage,
  saveItemInLocalStorage,
  removeItemFromLocalStorage,
  thisWeek,
  getFirstDayOfTheMonth,
  getTodaysDate,
};
