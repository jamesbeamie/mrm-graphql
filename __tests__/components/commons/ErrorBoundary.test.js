import React from 'react';
import { mount } from 'enzyme';
import ErrorBoundary from '../../../src/components/commons/ErrorBoundary';
import Spinner from '../../../src/components/commons/Spinner';
import { getItemFromLocalStorage } from '../../../src/utils/Utilities';

describe('ErrorBoundary component', () => {
  const wrapper = mount(
    <ErrorBoundary >
      <Spinner />
    </ErrorBoundary>,
  );
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should render children', () => {
    expect(wrapper.find('.spinner').length).toBe(1);
  });
  it('handles componentDidCatch()', () => {
    expect(wrapper.instance().componentDidCatch());
  });
  it('generates error message when an error is caught', () => {
    wrapper.setState({
      error: ' This is the error.',
      errorInfo: 'This is the error info',
    });
    expect(wrapper.text()).toEqual('Something went wrong');
  });

  it('it should remove clear cookies upon receiving invalid token', () => {
    const errorWrapper = mount(
      <ErrorBoundary isAuthError >
        <Spinner />
      </ErrorBoundary>,
    );
    errorWrapper.setState({
      error: ' This is the error.',
      errorInfo: 'This is the error info',
    });
    window.location.assign = jest.fn();
    global.localStorage = undefined;
    expect(getItemFromLocalStorage('key')).toEqual(null);
  });
});
