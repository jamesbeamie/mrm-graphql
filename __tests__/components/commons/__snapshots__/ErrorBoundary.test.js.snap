// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary component should render correctly 1`] = `
<ErrorBoundary
  error="Something went wrong"
  isAuthError={false}
>
  <Spinner
    size=""
  >
    <div
      className="centered"
    >
      <div
        className="spinner"
      />
    </div>
  </Spinner>
</ErrorBoundary>
`;
